packages/app-desktop/gui/JoplinCloudConfigScreen.tsx

```diff
@@ -17,8 +21,29 @@ const JoplinCloudConfigScreen = (props: JoplinCloudConfigScreenProps) => {
 
 	const isEmailToNoteAvailableInAccount = props.joplinCloudAccountType !== 1;
 
+	const goToJoplinCloudProfile = async () => {
+		await bridge().openExternal(`${props.joplinCloudWebsite}/users/me`);
+	};
```

Comment:
We should probably default to using `useCallback` for event handlers, especially when it's dependent on a prop like here. Probably the prop won't change but if for some reason it's initially empty, then set to a value, the component will incorrectly use the empty value

Reply:
> but if for some reason it's initially empty, then set to a value, the component will incorrectly use the empty value

I'm not sure that's true... I think `useCallback` [prevents unnecessary re-renders of child components](https://react.dev/reference/react/useCallback#reference) (rather than forcing the function to reload). More specifically, suppose that `props.joplinCloudWebsite` changes from `""` to `"https://joplincloud.com/"`:
1. `props.joplinCloudWebsite` is initially `""`. This is the initial render.
    - `goToJoplinCloudProfile` is set to a function. `props.joplinCloudWebsite` is `""` in its scope. As such, if `goToJoplinCloudProfile` is called, it will `openExternal('/users/me')`.
    - The component renders. A `Button` is given `goToJoplinCloudProfile` as a prop. This is the first render of that `Button`.
2. `props.joplinCloudWebsite` changes to `"https://joplincloud.com/"`. This causes the component to re-render.
    - `goToJoplinCloudProfile` is set to a function. `props.joplinCloudWebsite` is `"https://joplincloud.com/"` in its scope. As such, if `goToJoplinCloudProfile` is called, it will `openExternal('https://joplincloud.com//users/me')`.
    - The component renders. A `Button` is given `goToJoplinCloudProfile` as a prop. This prop is different from its previous value (because `function(){} !== function(){}`). As a result, the `Button` re-renders.
        - This re-render means that the button will use the new version of `goToJoplinCloudProfile`.


Observe that in step 2, because React considers different functions to be unequal, the `Button` uses the new version of the function that includes `"https://joplincloud.com/"` for `props.joplinCloudWebsite` in its scope.


Also see [react.dev's "Should you add useCallback everywhere"](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere).

Reply:
Ah yes, that's true, it will re-render anyway.

But I actually think that it's reasonable to default to adding useCallback rather than wondering whether it needs to be optimised or not. In this case maybe it doesn't make a difference that it re-render even when the prop doesn't change, but maybe it does, who knows. That's the idea of mostly defaulting to using useCallback - then we don't need to wonder about this, and it feels like it's more future proof